frontstage/controllers/party_controller.py

```diff
@@ -229,32 +230,112 @@ def get_respondent_enrolments(party_id):
                 }
 
 
+class CachingDataThreadCreation(threading.Thread):
```

Comment:
I am not sure of the advantage of using a class to launch the thread as opposed to launching directly. Both will work here .Also this is now a general purpose thread wrapper. So its not tied to anything to do with caching . Shouldn't it be in a file of its own in in the common folder with a more generic class name and a Docstring to reflect its use as a generic thread wrapper

Reply:
It's a generic class that can take in any function with any amount of arguments so it makes sense to move it to the common folder. I'll change the name to ThreadWrapper 👍